cleverhans/attacks_eager.py

```diff
@@ -0,0 +1,331 @@
+from abc import ABCMeta
+import numpy as np
+from six.moves import xrange
+import warnings
+import collections
+import tensorflow as tf
+from distutils.version import LooseVersion
+import cleverhans.utils as utils
+
+_logger = utils.create_logger("cleverhans.attacks_eager")
+
+if LooseVersion(tf.__version__) < LooseVersion('1.8.0'):
+    error = ('For eager execution',
+             'use Tensorflow version greather than 1.8.0.')
+    print(error)
+    exit(1)
+
+try:
+    from cleverhans.model_eager import Model_Eager
+except ImportError:
+    error = 'Import Error: cleverhans.model_eager'
+    raise ImportError(error)
+
+
+class Attack(object):
+    """
+    Abstract base class for all attack classes.
+    """
+    __metaclass__ = ABCMeta
+
+    def __init__(self, model, back='tfe', dtypestr='float32'):
+        """
+        :param model: An instance of the cleverhans.model.Model_Eager class.
+        :param back: The backend to use. Currently 'tfe' is the only option.
+        :param dtypestr: datatype of the input data samples and crafted
+                        adversarial attacks.
+        """
+        # Validate the input arguments.
+        if not(back == 'tfe'):
```

Comment:
A simpler way to do it is
if back != 'tfe':

Reply:
True, will update PR to reflect this change.